mono/profiler/proflog.c

```diff
@@ -1022,19 +1029,26 @@ write_int64 (char *buf, int64_t value)
 static void
 dump_header (MonoProfiler *profiler)
 {
-	char hbuf [128];
+	char hbuf [256];
```

Comment:
This will overflow if the extra data it too big.

All it take is passing "output=/very/log/path" to the profiler and we're busted.

You should calculate the size and alloc a big enough buffer.